### PR TITLE
Use session_for_view and sessions_for_view less

### PR DESCRIPTION
**Pull request by rwols**
Instead use methods of the LSPViewEventListener.

### CODE REVIEW

File: plugin/core/windows.py [hunk 1]
```diff
@@ -350,11 +350,13 @@ def window(self) -> sublime.Window:
 
     def sessions(self, view: sublime.View, capability: Optional[str] = None) -> Generator[Session, None, None]:
         file_name = view.file_name() or ''
+        scope = view2scope(view)
         for sessions in self._sessions.values():
             for session in sessions:
-                if capability is None or capability in session.capabilities:
-                    if session.state == ClientStates.READY and session.handles_path(file_name):
-                        yield session
+                if session.config.match_document(scope):
```
Comment (rchl) [inline plugin/core/windows.py:356]: Style nit: Could filter the list in the `for` loop already? I don't like a lot of nesting. :)

File: plugin/color.py [hunk 2]
```diff
@@ -35,10 +34,7 @@ def on_activated_async(self) -> None:
             self.initialize()
 
     def initialize(self, is_retry: bool = False) -> None:
-        if not any(configurations_for_view(self.view)):
-            self.initialized = True  # no server enabled, re-open file to activate feature.
-        sessions = list(sessions_for_view(self.view, 'colorProvider'))
-        if sessions:
+        if self.session('colorProvider'):
```
Comment (rchl) [inline plugin/color.py:37]: Style nit: I dislike such generic method names as it's really hard to search for uses of it in the code (especially when refactoring code). Could we maybe call those `get_session` and `get_sessions`?

File: plugin/core/types.py [hunk 3]
```diff
@@ -197,3 +197,4 @@ def views(self) -> List[ViewLike]:
 
     def run_command(self, command_name: str, command_args: Dict[str, Any]) -> None:
         ...
+        ...
```
Comment (rchl) [inline plugin/core/types.py:200]: Intentional?

File: plugin/color.py (hunk 2 above)
Comment (rchl) [inline plugin/color.py:37]: The logic has changed here a bit. Previously `self.initialized = True` was set if no sessions were found. Probably not a big deal. Someone would have to activate file twice really quickly for this to make a difference.

File: plugin/core/types.py (hunk 3 above)
Comment (rwols) [reply plugin/core/types.py]: No :) must have been pyls going crazy with format-on-save again

File: plugin/core/windows.py (hunk 1 above)
Comment (rwols) [reply plugin/core/windows.py]: I've put all of the checks in `can_handle` of `Session`. Hope that makes it a bit better. I don't personally mind nested ifs.

File: plugin/core/windows.py [hunk 4]
```diff
@@ -349,14 +349,10 @@ def window(self) -> sublime.Window:
...
     def sessions(self, view: sublime.View, capability: Optional[str] = None) -> Generator[Session, None, None]:
-        file_name = view.file_name() or ''
-        scope = view2scope(view)
         for sessions in self._sessions.values():
             for session in sessions:
-                if session.config.match_document(scope):
-                    if capability is None or capability in session.capabilities:
-                        if session.state == ClientStates.READY and session.handles_path(file_name):
-                            yield session
+                if session.can_handle(view, capability):
```
Comment (rchl) [inline plugin/core/windows.py]: Could still be filtered in the `for` loop. :)
`for session in [session for session in sessions if session.can_handle(view, capability)]:`

File: plugin/color.py (hunk 2 above)
Comment (rchl) [reply plugin/color.py:37]: I've approved but I don't like it.

Comment (rwols) [reply plugin/color.py:37]: I'll do it in another PR. This is currently more in line with the `self.session` from LspTextCommand.

**Pull request merged**